server/dynamo.go

```diff
@@ -108,3 +154,52 @@ func (c *Server) redeemTokenWithDynamo(issuer *Issuer, preimage *crypto.TokenPre
 	}
 	return nil
 }
+
+// CheckRedeemedTokenEquivalence returns whether just the ID of a given RedemptionV2 token
+// matches an existing persisted record, the whole value matches, or neither match and
+// this is a new token to be redeemed.
+func (c *Server) CheckRedeemedTokenEquivalence(issuer *Issuer, preimage *crypto.TokenPreimage, payload string, offset int64) (*RedemptionV2, Equivalence, *utils.ProcessingError) {
```

Comment:
Is there a reason we are using `*utils.ProcessingError` instead of `error`?  We can `errors.As` a `ProcessingError` in the caller?  I think this ProcessingError pointer might cause us issues down the line.

Reply:
Very reasonable. I've made the change.